src/commands/lambda/flare.ts

```diff
@@ -40,105 +49,170 @@ export class LambdaFlareCommand extends Command {
+    this.context.stdout.write(`\n${configStr}\n`)
 
-    // Print config
-    const configStrColored = util.inspect(config, false, undefined, true)
-    this.context.stdout.write(`\n${configStrColored}\n`)
-    if (this.isDryRun) {
-      this.context.stdout.write('\n🚫 The configuration was not sent as it was executed in dry run mode.\n')
+    // Get CloudWatch logs
+    const logs: [string, OutputLogEvent[]][] = []
+    if (this.withLogs) {
+      this.context.stdout.write('\n☁️ Getting CloudWatch logs...\n')
+      const cwlClient = new CloudWatchLogsClient({region})
+      const functionName = this.functionName.startsWith('arn:aws') ? this.functionName.split(':')[6] : this.functionName
+      const logGroupName = `/aws/lambda/${functionName}`
+      let logStreamNames
+      try {
+        logStreamNames = await getLogStreamNames(cwlClient, logGroupName)
+      } catch (err) {
+        this.context.stderr.write(commonRenderer.renderError(`Unable to get log streams: ${err.message}`))
+
+        return 1
+      }
+      if (logStreamNames === undefined) {
+        this.context.stdout.write(
+          commonRenderer.renderSoftWarning('No CloudWatch streams were found. Logs will not be retrieved or sent.\n')
+        )
+      } else {
+        this.context.stdout.write(`\n✅ Found log streams:\n• ${logStreamNames.join('\n• ')}\n\n`)
+        for (const logStreamName of logStreamNames) {
+          let logEvents
+          try {
+            logEvents = await getLogEvents(cwlClient, logGroupName, logStreamName)
+          } catch (err) {
+            this.context.stderr.write(
+              commonRenderer.renderError(`Unable to get log events for stream ${logStreamName}: ${err.message}`)
+            )
 
-      return 0
+            return 1
+          }
+          if (logEvents === undefined) {
+            this.context.stdout.write(
+              commonRenderer.renderSoftWarning(`No CloudWatch logs found in stream ${logStreamName}. Skipping...`)
+            )
+            continue
+          }
+          logs.push([logStreamName, logEvents])
+        }
+      }
     }
 
-    // Send data to Datadog
-    const folderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
-    const filePath = path.join(folderPath, FUNCTION_CONFIG_FILE_NAME)
-    const zipPath = path.join(folderPath, ZIP_FILE_NAME)
-    const configStrUncolored = JSON.stringify(config, undefined, 2)
+    this.context.stdout.write('\n💾 Saving files...\n')
     try {
-      await writeFile(folderPath, filePath, configStrUncolored, this.context)
-      await zipContents(filePath, zipPath)
+      // Create folders
+      const rootFolderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
+      const logsFolderPath = path.join(rootFolderPath, LOGS_DIRECTORY)
+      if (fs.existsSync(rootFolderPath)) {
+        deleteFolder(rootFolderPath)
+      }
+      createDirectories(rootFolderPath, logsFolderPath, logs.length > 0)
```

Comment:
I think this logic is too complex.

1. The logs folder can be calculated inside `createDirectores`, because all its dependencies are available inside the function.
2. Wouldn't it make sense to just send the logs instead of a boolean? If you are going to check either way if it's empty, you can do that inside the function. 

Wdyt?

Reply:
I see, I didn't catch that it was being used again. Thanks for clarifying!